Trappist/src/Promact.Trappist.Web/wwwroot/app/profile/profile-dashboard/change-password-dialog.component.ts

```diff
@@ -1,8 +1,46 @@
 ﻿import { Component } from '@angular/core';
+import { ChangePasswordModel } from '../password.model';
+import { MdDialogRef } from '@angular/material';
+import { ProfileService } from '../profile.service';
+import { MdSnackBar } from '@angular/material';
 
 @Component({
-  moduleId: module.id,
-  selector: 'change-password-dialog',
-  templateUrl: 'change-password-dialog.html'
+    moduleId: module.id,
+    selector: 'change-password-dialog',
+    templateUrl: 'change-password-dialog.html'
 })
-export class ChangePasswordDialogComponent { }
+export class ChangePasswordDialogComponent {
+    constructor(public profileService: ProfileService, public dialog: MdDialogRef<any>, public snackBar: MdSnackBar) { }
+    user: ChangePasswordModel = new ChangePasswordModel();
+    isMessageMismatch: boolean;
+    messageMismatch: string;
+    response: any;
+    errorMesseage: any;
+    errorCorrection: boolean = true;
+
+    /**
+     * update the database with new password
+     * @param userPassword of type ChangePasswordModel which has the new password of the user
+     */
+    changePassword(userPassword: ChangePasswordModel) {
+        if (userPassword.newPassword === userPassword.confirmPassword) {
+            this.isMessageMismatch = true;
```

Comment:
isMessageMismatch = true; when we read, it fills like there is a mismatch.  Ideally, it should be false. change true-false assignment as per  the variable name, or change variable name